lib/Index/Index.cpp

```diff
@@ -1504,6 +1504,13 @@ bool IndexSwiftASTWalker::reportExtension(ExtensionDecl *D) {
   if (!startEntity(D, Info, /*IsRef=*/false))
     return false;
 
+  if (auto *DR = dyn_cast_or_null<DeclRefTypeRepr>(D->getExtendedTypeRepr()))
```

Comment:
I assume there's a better way to do this that I hope someone can advise me on

Reply:
Rather than `getNominalType` above, could we grab `getExtendedType` and then use `reportRelatedTypeRef` instead of `reportRelatedRef` which would then also handle adding the implicit reference to the underlying NTD?

Reply:
Nice! I updated to do this but I don't think I could entirely remove the `getNominalType` call since other functions require that? Am I missing something there?

Reply:
Not missing anything, I just didn't notice it being used elsewhere when I was looking at it :).